pkgs/applications/graphics/glimpse/plugins/default.nix

```diff
@@ -0,0 +1,260 @@
+# preferences -> Folders -> Plug-ins
+# same applies for the scripts
+
+{ config, pkgs, glimpse }:
+let
+  inherit (pkgs) stdenv fetchurl pkgconfig intltool glib fetchFromGitHub;
+  inherit (glimpse) targetPluginDir targetScriptDir;
+
+  pluginDerivation = a: let
+    name = a.name or "${a.pname}-${a.version}";
+  in
+    stdenv.mkDerivation (
+      {
+        prePhases = "extraLib";
+        extraLib = ''
+          installScripts(){
+            mkdir -p $out/${targetScriptDir}/${name};
+            for p in "$@"; do cp "$p" -r $out/${targetScriptDir}/${name}; done
+          }
+          installPlugins(){
+            mkdir -p $out/${targetPluginDir}/${name};
+            for p in "$@"; do cp "$p" -r $out/${targetPluginDir}/${name}; done
+          }
+        '';
+      }
+      // a
+      // {
+        name = "glimpse-plugin-${name}";
+        buildInputs = [ glimpse glimpse.gtk glib ] ++ (a.buildInputs or []);
+        nativeBuildInputs = [ pkgconfig intltool ] ++ (a.nativeBuildInputs or []);
+      }
+    );
+
+  scriptDerivation = { src, ... }@attrs: pluginDerivation (
+    {
+      phases = [ "extraLib" "installPhase" ];
+      installPhase = "installScripts ${src}";
+    } // attrs
+  );
+
+in
+
+stdenv.lib.makeScope pkgs.newScope (
+  self: with self; {
+    gap = pluginDerivation {
+      /* menu:
+         Video
+      */
+      name = "gap-2.6.0";
+      src = fetchurl {
+        url = https://ftp.gimp.org/pub/gimp/plug-ins/v2.6/gap/gimp-gap-2.6.0.tar.bz2;
+        sha256 = "1jic7ixcmsn4kx2cn32nc5087rk6g8xsrz022xy11yfmgvhzb0ql";
+      };
+      NIX_LDFLAGS = [ "-lm" ];
+      patchPhase = ''
+        sed -e 's,^\(GIMP_PLUGIN_DIR=\).*,\1'"$out/${glimpse.name}-plugins", \
+         -e 's,^\(GIMP_DATA_DIR=\).*,\1'"$out/share/${glimpse.name}", -i configure
+      '';
+      hardeningDisable = [ "format" ];
+      meta = with stdenv.lib; {
+        description = "The GIMP Animation Package";
+        homepage = https://www.gimp.org;
+        # The main code is given in GPLv3, but it has ffmpeg in it, and I think ffmpeg license
+        # falls inside "free".
+        license = with licenses; [ gpl3 free ];
+      };
+    };
+
+    fourier = pluginDerivation rec {
+      /* menu:
+         Filters/Generic/FFT Forward
+         Filters/Generic/FFT Inverse
+      */
+      name = "fourier-0.4.3";
+      buildInputs = with pkgs; [ fftw ];
+
+      src = fetchurl {
+        url = "https://www.lprp.fr/files/old-web/soft/gimp/${name}.tar.gz";
+        sha256 = "0mf7f8vaqs2madx832x3kcxw3hv3w3wampvzvaps1mkf2kvrjbsn";
+      };
+
+      installPhase = "installPlugins fourier";
+
+      meta = with stdenv.lib; {
+        description = "GIMP plug-in to do the fourier transform";
+        homepage = "https://people.via.ecp.fr/~remi/soft/gimp/gimp_plugin_en.php3#fourier";
+        license = with licenses; [ gpl3Plus ];
+      };
+    };
+
+    focusblur = pluginDerivation rec {
```

Comment:
We should probably remove the broken plug-ins at least.

Reply:
Makes sense. I removed `focusBlur`, `texturize` and `exposureBlend`.